TestingScripts/Coverage/CoverageOpenCPP.py

```diff
@@ -0,0 +1,764 @@
+        else:
+            wsSumm.write(endRow, startCol + 5, 0.0, rp.percent2)
+        wsSumm.write(endRow, startCol + 6, TotalFiles - CoveredFiles, rp.num2)
+
+        wsSumm.write(endRow, startCol + 7, "", rp.border)
+        wsSumm.write(endRow, startCol + 8, TotalLines , rp.num2)
+        wsSumm.write(endRow, startCol + 9, CoveredLines, rp.num2)
+        if float(TotalFiles) > 0.0 and float(TotalLines) > 0.0:
+            wsSumm.write(endRow, startCol + 10, (float(CoveredLines) / float(TotalLines)), rp.percent2)
+        else:
+            wsSumm.write(endRow, startCol + 10, 0.0, rp.percent2)
+        wsSumm.write(startRow + i, startCol + 11, TotalLines - CoveredLines, rp.num2)
+
+        wsSumm.write(endRow, startCol + 12, "", rp.border)
+        wsSumm.write(endRow, startCol + 13, TotalMethods, rp.num2)
+        wsSumm.write(endRow, startCol + 14, CoveredMethods, rp.num2)
+        if float(TotalFiles) > 0.0 and float(TotalMethods) > 0.0:
+            wsSumm.write(endRow, startCol + 15, (float(CoveredMethods) / float(TotalMethods)), rp.percent2)
+        else:
+            wsSumm.write(endRow, startCol + 15, 0.0, rp.percent2)
+        wsSumm.write(endRow, startCol + 16, TotalMethods - CoveredMethods, rp.num2)
+        wsSumm.write(endRow, startCol + 17, "", rp.num2)        
+        wsSumm.write(endRow, startCol + 18, "", rp.leftborder)
+
+        #Now add high level summary
+        # self.addHighSummary(rp, wsHigh)  TODO: will be done when the feature of method coverage will be integrated
+        self.addHighSummary(rp, wsHigh2, False)
+    #-------------------------------------------------------------------------------------------
+    # bsimethod
+    #-------------------------------------------------------------------------------------------
+    def methodsHtml(self):
+        
+        print ('\n*** writing methods to html \n')
+        self.runCoverage()
+        self.methodsFromPDBs()
+        self.coverageAll()
+
+        for comp in self.comps:
+            methods = self.comps[comp]['methods']
+            compName = cmp.GetName(comp)
+            exeName = os.path.basename(cmp.ExePathForComp(comp))
+            moduleFile = os.path.join(self.repoPath, exeName +'.html', 'Modules', cmp.DllForComp(comp)[:-4]+'_methods.html')
+            html_file = open(moduleFile,"w")
+            html_file.write('<html><body>')
+            html_file.write('<h1>')
+            html_file.write(exeName)
+            html_file.write('</h1>')
+            html_file.write('<table border="1">')
+            html_file.write('<tr><th>')
+            html_file.write('Method Name')
+            html_file.write('</th><th>')
+            html_file.write('File Name')
+            html_file.write('</th></tr>')
+
+            total = len(methods)
+            covered = 0
+            f = open(os.path.join(self.repoPath, exeName+'.html', 'methods.html'),"w")
+            for m in methods:
+                if methods[m]['covered']:
```

Comment:
I think this section could be condensed down slightly. Is it possible to set the bgcolor to a variable based off of whether or not the method is covered and then you would only need to have all of the html_file.write stuff once between the `if methods[m]['covered']` and it's else?

Reply:
removed

Reply:
same concerns as above

Reply:
done